connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -264,12 +279,62 @@ public void startConnector(
 
                 log.info("Creating connector {} of type {}", connName, connClass);
                 final Connector connector = plugins.newConnector(connClass);
-                final ConnectorConfig connConfig = ConnectUtils.isSinkConnector(connector)
-                        ? new SinkConnectorConfig(plugins, connProps)
-                        : new SourceConnectorConfig(plugins, connProps, config.topicCreationEnable());
+                final ConnectorConfig connConfig;
+                final CloseableOffsetStorageReader offsetReader;
+                if (ConnectUtils.isSinkConnector(connector)) {
+                    connConfig = new SinkConnectorConfig(plugins, connProps);
+                    offsetReader = null;
```

Comment:
Ah, good catch! I think this is one more place where `Utils::closeQuietly` comes in handy since it does the null check for us, won't interrupt shutdown if an exception is thrown, and further standardizes the cleanup logic for the code base.

Alternatively, we could retain the current behavior, but I don't see much use in constructing an offset reader for sink connectors that won't be used now and is unlikely to be used in the future.